contents/newsletter/feature-flag-mistakes.md

```diff
@@ -0,0 +1,178 @@
+---
+title: Don't make these classic feature flag mistakes
+date: 2025-01-08
+author:
+ - ian-vanagas
+featuredImage: >-
+  https://res.cloudinary.com/dmukukwp6/image/upload/posthog.com/contents/images/blog/gitlab-github-flags.jpeg
+featuredImageType: full
+tags:
+  - Engineering
+crosspost:
+  - Product engineers
+  - Founders
+  - Blog
+---
+
+Product engineers succeed when they can ship quickly, understand if a change is working, and iterate from there. Because they let you ship, deploy, and release software safely, [feature flags](/feature-flags) are a powerful tool for doing this.
+
+But with great power comes great responsibility. Misused feature flags have caused downtime for the largest apps in the world, created a mountain of tech debt, and nearly bankrupted a company (we'll cover this later). Luckily, these are all avoidable mistakes we can learn from and avoid.
+
+They boil down to two main issues:
+
+1. **Complexity creates confusion.** Embed and interlinked flags create many potential states that can quickly become untestable and unpredictable.
+
+2. **Poor maintainability creates tech debt.** The longer flags stick around the more likely they are to get reused in unintended ways or negatively react with new changes. 
+
+Both create situations where mistakes and unintended consequences can happen. Avoiding them means avoiding the issues we've put together here:
+
+## 1. Not using them
+
+Before I tell you the ways flags can go wrong, let me start by saying that the biggest potential mistake of them all is **not using feature flags**. 
+
+At [PostHog](/feature-flags) (and many other companies), they provide two main benefits:
+
+1. **Safety.** Flags enable progressive rollouts and kill switch rollbacks enabling you to deploy and release new code safely.
+
+2. **Speed.** By [separating deployment from release](/product-engineers/decouple-deployment-from-release), they enable us to do [trunk-based development](/product-engineers/trunk-based-development) meaning we can [ship more changes faster](https://newsletter.posthog.com/p/how-to-design-your-company-for-speed). 
+
+On top of this, they let us run [A/B tests](/experiments), [beta programs](/tutorials/public-beta-program), and [dogfood](/product-engineers/dogfooding) our changes. As a [remote and async team](https://newsletter.posthog.com/p/how-we-work-asynchronously), this is critical to our success. Our tactics for developing and releasing software would be a lot more limited without them.
```

Comment:
- This is a bit of a nit, but I think this section should talk more about the benefits to "you" (the reader) rather than us. We can still talk about how we use them, obviously.

- Random thought: Should we try this as the last point? I feel like the "let me start by saying that the biggest potential mistake of them all is **not using feature flags" kind of takes me out of the article a bit. This might work better there.